### PR TITLE
feat(ims): support `cbr_whole_image` resource

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:

Support `cbr_whole_image` resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

Support `cbr_whole_image` resource

```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/ims" TESTARGS="-run TestAccCbrWholeImage_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims -v -run TestAccCbrWholeImage_basic -timeout 360m -parallel 4
=== RUN   TestAccCbrWholeImage_basic
=== PAUSE TestAccCbrWholeImage_basic
=== CONT  TestAccCbrWholeImage_basic
--- PASS: TestAccCbrWholeImage_basic (776.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       776.722s

```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
image

### CODE REVIEW

File: huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		BackupId:            d.Get("backup_id").(string),
+		Description:         d.Get("description").(string),
+		MaxRam:              d.Get("max_ram").(int),
+		MinRam:              d.Get("min_ram").(int),
+		ImageTags:           imageTags,
+		EnterpriseProjectID: cfg.GetEnterpriseProjectID(d),
+		WholeImageType:      "CBR",
+	}
+
+	createResp, err = cloudimages.CreateWholeImageByServer(client, createOpts).ExtractJobResponse()
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go]:
```suggestion
	createResp, err := cloudimages.CreateWholeImageByServer(client, createOpts).ExtractJobResponse()
```

File: huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+	// If the list API return empty, then process `CheckDeleted` logic.
+	if len(imageList) < 1 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "IMS CBR whole image")
+	}
+
+	image := imageList[0]
+	imageTags := getImageTags(d, client)
+	if err != nil {
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go]: Remove this `err` conditional judgment.

File: huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go (hunk 1 above)
Comment (jinyangyang222) [reply huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go]: fixed.

File: huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go (hunk 2 above)
Comment (jinyangyang222) [reply huaweicloud/services/ims/resource_huaweicloud_ims_cbr_whole_image.go]: fixed.

**Pull request merged**